### PR TITLE
[NETPATH-235][Network Path] Refactor Protocol and fix protocol in metric tag

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

[NETPATH-235][Network Path] Refactor Protocol and fix protocol in metric tag

### Describe how to test/QA your changes

[NETPATH-235]: https://datadoghq.atlassian.net/browse/NETPATH-235?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AyyLam — APPROVED** [review]

File: comp/networkpath/npcollector/npcollectorimpl/npcollector_test.go [hunk 1]
```diff
@@ -457,7 +457,7 @@ func Test_npCollectorImpl_ScheduleConns(t *testing.T) {
 				},
 			},
 			expectedPathtests: []*common.Pathtest{
-				{Hostname: "10.0.0.4", Port: uint16(80), Protocol: "tcp", SourceContainerID: "testId3"},
+				{Hostname: "10.0.0.4", Port: uint16(80), Protocol: "TCP", SourceContainerID: "testId3"},
```
Comment (ken-schneider) [inline comp/networkpath/npcollector/npcollectorimpl/npcollector_test.go:460]: Should this and the above test be `payload.TCP`?

**Review by ken-schneider — APPROVED** [review]
🥜  one general question I'm thinking about: would it make sense to move the definition of `Protocol` from `payload.go` to `common/pathtest.go`? The protocol seems like it's a property of the pathtest itself rather than the payload

File: comp/networkpath/npcollector/npcollectorimpl/npcollector_test.go (hunk 1 above)
Comment (AlexandreYang) [reply comp/networkpath/npcollector/npcollectorimpl/npcollector_test.go]: indeed, fixed

**Review by daniel-taf — APPROVED** [review]

**Comment by ken-schneider** [comment]
Ran QA using both static paths and dynamic paths. Was able to validate the changes both in the traces sent to the back end as well as the tagging of metrics.

**Pull request merged**